code/clue.js

```diff
@@ -1,59 +1,319 @@
+  weight: 0.2,
+  id: 'poison'
+}
+
+const axe = {
+  name: 'axe',
+  weight: 7,
+  id: 'axe'
+}
+
+const bat = {
+  name: 'bat',
+  weight: 3,
+  id: 'bat'
+}
+
+const trophy = {
+  name: 'trophy',
+  weight: 15,
+  id: 'trophy'
+}
+
+const pistol = {
+  name: 'pistol',
+  weight: 2,
+  id: 'pistol'
+}
+
+// ROOM VARIABLES
+
+const roomDining = 'Dining Room';
+
+const roomConservatory = 'Conservatory';
+
+const roomKitchen = 'Kitchen';
+
+const roomStudy = 'Study';
+
+const roomLibrary = 'Library';
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const roomBilliard = 'Billiard Room';
+
+const roomLounge = 'Lounge';
+
+const roomBallroom = 'Ballroom';
+
+const roomHall = 'Hall';
+
+const roomSpa = 'Spa';
+
+const roomLiving = 'Living Room';
+
+const roomObservatory = 'Observatory';
+
+const roomTheather = 'Theater';
+
+const roomGuest = 'Guest House';
+
+const roomPatio = 'Patio';
+
+// OBJECT THAT KEEPS THE MYSTERY.
+
+let mystery = {
+  killer: '',
+  weapon: '',
+  room: ''
+}
+
+// ARRAYS OF ALL SUSPECTS, WEAPONS AND ROOMS
 
 const suspects = [
   mrGreen,
+  prPlum,
+  msScarlet,
+  mrsPeacock,
+  colMustard,
   mrsWhite
-  // ...  and the rest
 ]
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
+
+const rooms = [
+  roomDining,
+  roomConservatory,
+  roomKitchen,
+  roomStudy,
+  roomLibrary,
+  roomBilliard,
+  roomLounge,
+  roomBallroom,
+  roomHall,
+  roomSpa,
+  roomLiving,
+  roomObservatory,
+  roomTheather,
+  roomGuest,
+  roomPatio
+]
 
-const rooms = []
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
+// FUNCTION THAT RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT PASSES IN TO THE FUNCTION.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// Function to shuffle the suspects favorite weapon
+const shuffleFavoriteWeapon = () => {
+  favoriteWeapon = weapons[Math.floor(Math.random() * weapons.length)]; //Picks a random weapon from the array and creates a favoriteWeapon object
+  mystery.killer.favoriteWeapon = favoriteWeapon.name;// Assigns the name of favoriteWeapon to choosen killers object
+  document.getElementById('favoriteWeapon').innerHTML = `Prefered weapon: ${mystery.killer.favoriteWeapon}`; //Prints favoriteweapon to card
+}
 
-// This function will be invoked when you click on the killer card.
+// This function will be invoked in the loadingPickKiller function.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  document.getElementById("loader-wrapper-killer").style.display = "none";
```

Comment:
It's better to use class manipulation, not style manipulation

Reply:
do you mean using querySelector or for ex use .className to add another class? Will the later override the previous values in the css? Or should I make two different classes for each? Not really sure how to best approach this :)

Reply:
Sorry, my comment might have been not clear enough :)
What I meant was that we should rather create class `hidden` like this :

```
.hidden {
    display: none;
}
```

and add it to your `loader-wrapper-killer` element in JS like that :

```
document.getElementById("loader-wrapper-killer").classList.add = "hidden";
```

Main idea behind that approach is that we should rather add or remove CSS classes instead of adding and removing exact styles

Reply:
Ok thanks for the explanation, I will try this! 😄